File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s452855095_commented.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&arr[i])
	}
	cnt := 0
	flag := true
	for {
		for i := 0; i < n; i++ {
			if arr[i]%2 != 0 {
				flag = false
				break
			} else {
				arr[i] = arr[i] / 2
			}
		}
		if flag {
			cnt++
		} else {
			break
		}
		flag = true
	}
	fmt.Println(cnt)
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/C++/s452855095_commented.cpp
#include <iostream>
#include <vector>

int main() {
    std::cin.tie(nullptr);
    std::ios::sync_with_stdio(false);

    int n;
    std::cin >> n;
    std::vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> arr[i];
    }
    int cnt = 0;
    bool flag = true;
    while (true) {
        flag = true;
        for (int i = 0; i < n; ++i) {
            if (arr[i] % 2 != 0) {
                flag = false;
                break;
            } else {
                arr[i] /= 2;
            }
        }
        if (flag) {
            cnt++;
        } else {
            break;
        }
    }
    std::cout << cnt;
    return 0;
}
// End of Code
